http3/request_writer.go

```diff
@@ -70,20 +68,15 @@ func (w *requestWriter) WriteRequest(str quic.Stream, req *http.Request, gzip bo
 			}
```

Comment:
In line 61 above I think there's a bug: io.Reader allows returning n > 0 together with io.EOF in one, so we should only break if err == io.EOF && n == 0, and also fix the CancelWrite handling.

(GH doesn't allow me to comment on a line that's not part of this commit 🙄

Reply:
Good catch! That's a separate bug though, I opened #2641 to track it.